core/src/main/scala/caliban/schema/Step.scala

```diff
@@ -10,12 +10,16 @@ import zio.stream.ZStream
 sealed trait Step[-R]
 
 object Step {
-  case class ListStep[-R](steps: List[Step[R]])                                    extends Step[R]
-  case class FunctionStep[-R](step: Map[String, InputValue] => Step[R])            extends Step[R]
-  case class MetadataFunctionStep[-R](step: Field => Step[R])                      extends Step[R]
-  case class ObjectStep[-R](name: String, fields: collection.Map[String, Step[R]]) extends Step[R]
-  case class QueryStep[-R](query: ZQuery[R, Throwable, Step[R]])                   extends Step[R]
-  case class StreamStep[-R](inner: ZStream[R, Throwable, Step[R]])                 extends Step[R]
+  case class ListStep[-R](steps: List[Step[R]])                         extends Step[R]
+  case class FunctionStep[-R](step: Map[String, InputValue] => Step[R]) extends Step[R]
+  case class MetadataFunctionStep[-R](step: Field => Step[R])           extends Step[R]
+  case class QueryStep[-R](query: ZQuery[R, Throwable, Step[R]])        extends Step[R]
+  case class StreamStep[-R](inner: ZStream[R, Throwable, Step[R]])      extends Step[R]
+
+  case class ObjectStep[-R](name: String, fields: String => Option[Step[R]]) extends Step[R]
```

Comment:
I was thinking we could potentially remove `Option` and return `NullStep` directly to save the allocation. But you won't be able to use `orElse`, even though it can be worked around. Any thoughts?

Reply:
Maybe it's okay since merging arbitrarily keeps one of them anyway (we're not supposed to have the same field twice in theory)?

Reply:
Yeah I think you're right. The chances that this is a legit usecase are extremely low anyways. Changed `fields` to `String => Step[R]` instead

Reply:
Squeezing the performance to the last bit 😄